acceptance/tests/external_facts/non_root_users_default_external_fact_directory.rb

```diff
@@ -0,0 +1,157 @@
+confine :except, :platform => 'eos-'    # does not support user creation ARISTA-37
+
+# TestRail: C64580
+
+require 'facter/acceptance/user_fact_utils'
+extend Facter::Acceptance::UserFactUtils
+
+# Generate an external fact dynamically
+def ext_user_fact(value='BASIC')
+  "test: '#{value}'"
+end
+
+# Retrieve a specific users home directory $HOME/.facter
+#
+def get_user_facter_dir(user_home, platform)
+  if platform =~ /windows/
+    raise "get_user_facter_dir: not a supported directory on Windows"
+  else
+    File.join(user_home, '.facter')
+  end
+end
+
+# Retrieve a specific users home facts directory $HOME/.facter/facts.d
+#
+def get_user_facts_dir(user_home, platform)
+  if platform =~ /windows/
+    raise "get_user_facts_dir: not a supported directory on Windows"
+  else
+    File.join(get_user_facter_dir(user_home, platform), 'facts.d')
+  end
+end
+
+# Retreive a specific users home puppetlabs directory $HOME/.puppetlabs
+#
+def get_user_puppetlabs_dir(user_home, platform)
+  if platform =~ /windows/
+    raise "get_user_puppetlabs_dir: not a supported directory on Windows"
+  else
+    File.join(user_home, '.puppetlabs')
+  end
+end
+
+# Retreive a specific users home puppetlabs facts directory $HOME/.puppetlabs/opt/facter/facts.d
+#
+def get_user_puppetlabs_facts_dir(user_home, platform)
+  if platform =~ /windows/
+    raise "get_user_puppetlabs_facts_dir: not a supported directory on Windows"
+  else
+    File.join(get_user_puppetlabs_dir(user_home, platform), 'opt', 'facter', 'facts.d')
+  end
+end
+
+# retrieve the users home directory for a host and user
+#
+def get_home_dir(host, user_name)
+  home_dir = nil
+  on host, puppet_resource('user', user_name) do |result|
+    home_dir = result.stdout.match(/home\s*=>\s*'([^']+)'/m)[1]
+  end
+  home_dir
+end
+
+# retrieve the correct shell path for system under test
+#
+def user_shell(agent)
+  if agent['platform'] =~ /windows/
+    raise "a user shell is not available on Windows"
+  elsif agent['platform'] =~ /aix/
+    '/usr/bin/bash'
+  else
+    '/bin/bash'
+  end
+end
+
+agents.each do |agent|
+  non_root_user = "nonroot"
+
+  step "Agent #{agent}: create a #{non_root_user} to run facter with"
+  on(agent, "puppet resource user #{non_root_user} ensure=present managehome=true shell='#{user_shell(agent)}'")
+
+  user_home = get_home_dir(agent, non_root_user)
+
+  # The directories that facter processes facts for a user from
+  user_base_facts_dir = get_user_facter_dir(user_home, agent['platform'])
+  user_facts_dir = get_user_facts_dir(user_home, agent['platform'])
+  user_facts_path = "#{user_facts_dir}/test.yaml"
+
+  user_base_puppetlabs_dir = get_user_puppetlabs_dir(user_home, agent['platform'])
+  user_puppetlabs_facts_dir = get_user_puppetlabs_facts_dir(user_home, agent['platform'])
+  user_puppetlabs_facts_path = "#{user_puppetlabs_facts_dir}/test.yaml"
+
+  step "Agent #{agent}: figure out facter program location"
+  facter_path = on(agent, "which facter").stdout.chomp
```

Comment:
Not sure it will matter here, but I'll point out that `which facter` will return the path to the first match. That's probably what we want, but it's no the only possible option. Facter might (wrongly) be installed in two locations, for example.